Owncloud iOs Client/Tabs/FileTab/Share/ShareMainViewController.m

```diff
@@ -616,232 +616,319 @@ - (BOOL)alertViewShouldEnableFirstOtherButton:(UIAlertView *)alertView
 
 - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
     
-    return shareTableViewSectionsNumber;
+    NSInteger numberOfSections = shareTableViewSectionsNumber;
+    
+    if (!k_is_share_with_users_available) {
+        numberOfSections--;
+    }
+    
+    if (!k_is_share_by_link_available) {
+        numberOfSections--;
+    }
+    
+    return numberOfSections;
 }
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
     
     if (section == 0) {
         return 1;
-    }else if (section == 1){
+    }else if (section == 1 && k_is_share_with_users_available){
         if (self.sharedUsersOrGroups.count == 0) {
            return self.sharedUsersOrGroups.count + 2;
         }else{
            return self.sharedUsersOrGroups.count + 1;
         }
-        
-    }else{
+    } else if (section == 1 && k_is_share_by_link_available){
```

Comment:
section == 1 && k_is_share_by_link_available option contains the same code section == 2 && k_is_share_by_link_available does